tests/contest/contest/src/utils/test_utils.rs

```diff
@@ -222,7 +222,12 @@ pub fn test_inside_container(
         .wait_with_output()
     {
         Ok(c) => c,
-        Err(e) => return TestResult::Failed(anyhow!("container start failed : {:?}", e)),
+        Err(e) => {
+            // given that start has failed, we can be pretty sure that create has either failed
+            // or completed already, so we wait on it so it does not become a zombie process
+            let _ = create_process.wait_with_output();
+            return TestResult::Failed(anyhow!("container start failed : {:?}", e));
```

Comment:
I would like someone to take another look at this, the clippy error was that in the early return codepath here, we do not wait on the create process, hence it can become a zombie process. Hence I added a wait here with assumption that if the `start_container` has failed, we can be fairly certain that the original create has also errored or has already exited.

However, would like someone else to confirm the logic.

Reply:
Why don't we create a different PR for only this part? This could be an important change.

Reply:
Hey, ok, I'll create a  separate PR just for this change, and once that is resolved I can rebase this PR.

Reply:
I have opened https://github.com/youki-dev/youki/pull/3130 to extract just this change, once that is merged, I can remove it from here if needed.